src/zarr/storage/_builtin_adapters.py

```diff
@@ -0,0 +1,562 @@
+        read_only = cls._determine_read_only_mode(preceding_url, **kwargs)
+
+        # Apply scheme-specific URL normalization
+        normalized_url = cls._normalize_url(preceding_url, scheme)
+
+        return FsspecStore.from_url(
+            normalized_url, storage_options=storage_options, read_only=read_only
+        )
+
+    @classmethod
+    def _is_remote_url(cls, url: str) -> bool:
+        """Check if URL is a supported remote URL."""
+        return "://" in url and not url.startswith("file:")
+
+    @classmethod
+    def _extract_scheme(cls, url: str) -> str:
+        """Extract scheme from URL."""
+        return url.split("://", 1)[0]
+
+    @classmethod
+    def _validate_scheme(cls, scheme: str) -> None:
+        """Validate that the scheme is supported."""
+        supported = cls.get_supported_schemes()
+        if scheme not in supported:
+            raise ValueError(
+                f"Unsupported scheme '{scheme}'. Supported schemes: {', '.join(supported)}"
+            )
+
+    @classmethod
+    def _determine_read_only_mode(cls, url: str, **kwargs: Any) -> bool:
+        """Determine read-only mode with scheme-specific defaults."""
+        # Check explicit mode in kwargs
+        if "mode" in kwargs:
+            return bool(kwargs["mode"] == "r")
+
+        # Check storage_options
+        storage_options = kwargs.get("storage_options", {})
+        if "read_only" in storage_options:
+            return bool(storage_options["read_only"])
+
+        # Scheme-specific defaults
+        scheme = cls._extract_scheme(url)
+        return scheme in ("http", "https")  # HTTP(S) typically read-only, others can be writable
+
+    @classmethod
+    def _normalize_url(cls, url: str, scheme: str) -> str:
+        """Apply scheme-specific URL normalization."""
+        return url
+
+    @classmethod
+    def can_handle_scheme(cls, scheme: str) -> bool:
+        return scheme in cls.get_supported_schemes()
+
+    @classmethod
+    def get_supported_schemes(cls) -> list[str]:
+        return [
+            "s3",
+            "gs",
+            "http",
+            "https",  # HTTP(S)
+            "abfs",
+            "az",  # Azure Blob Storage
+            "ftp",
+            "ftps",  # FTP
+            # Could be extended with more fsspec-supported schemes
+        ]
+
+
+class HttpsAdapter(RemoteAdapter):
+    """Store adapter for HTTP(S) URLs using fsspec."""
+
+    adapter_name = "https"
+
+    @classmethod
+    def get_supported_schemes(cls) -> list[str]:
+        return ["http", "https"]
+
+
+class S3Adapter(RemoteAdapter):
+    """Store adapter for S3 URLs using fsspec.
+
+    Supports:
+    - Standard AWS S3: s3://bucket/path
+    - Custom S3 endpoints: s3+http://endpoint/bucket/path, s3+https://endpoint/bucket/path
+    """
+
+    adapter_name = "s3"
+
+    @classmethod
+    def get_supported_schemes(cls) -> list[str]:
+        return ["s3", "s3+http", "s3+https"]
```

Comment:
for the `s3+` urls we aren't reaching here unfortunately:

```python
zarr.open(
  "s3+https://uk1s3.embassy.ebi.ac.uk/idr/zarr/v0.5/idr0062A/6001240_labels.zarr",
  storage_options={'anon': True}
)
```
```
ValueError: Protocol not known: s3+https
```

i think this relates to the prior mentioned issues with `is_zep8_url`